library/core/src/iter/adapters/fuse.rs

```diff
@@ -182,6 +184,9 @@ where
     }
 }
 
+#[unstable(feature = "trusted_len", issue = "37572")]
+unsafe impl<I> TrustedLen for Fuse<I> where I: TrustedLen {}
```

Comment:
Could you add a `// SAFETY:` comment?

Reply:
Indeed, also for the `TrustedRandomAccess` below. That `TrustedLen` is OK is a weaker requirement than `TrustedRandomAccess`

Reply:
Done